tests/stdlib/tstrmisc.nim

```diff
@@ -0,0 +1,22 @@
+import strmisc
+
+
+func main() =
+  doAssert parseFloatThousandSep("0") == 0.0
```

Comment:
are leading `0` supposed to work? eg `parseFloatThousandSep("001")` currently works

Reply:
Yes, feel free to send PR to `parseFloat` if you think otherwise.

```nim
doAssert parseFloat("0") == 0.0
doAssert parseFloat("001") == 1.0
```

Reply:
https://tio.run/##K6gsycjPM/7/v6AoM69EIy0nP7FEQ8lASVOTC1XEwBBDzNAADIDi//8DAA